RFS/src/main/java/org/opensearch/migrations/bulkload/common/BulkDocSection.java

```diff
@@ -34,16 +34,28 @@ public class BulkDocSection {
 
     @EqualsAndHashCode.Include
     @Getter
-    private final String docId;
+    private final int luceneSegId;
```

Comment:
Individual bulk docs might be a little too granular for reporting the subshard items.  What do you think about creating a BulkDocs object that replaces List<BulkDocSecion> which knows the start/end positions of the docs within?

Report updates to the subshard progress based on a range rather that list of indices seems cleaner.

Reply:
Yes, the comment is a broad design statement.  I'm not clear on how everything is going into the flux, but the current organization seems have _implicit_ relationships on processing order which break down if going from sequential to a non-sequental processing such as a batch that is 'stuck' on a single document that persists on failing to upload where as many other batches could keep being processed around it.

The background of this was built to represent the individual bulk request item to include a document which was part of in a list where ordering was not relevant, since then it has been shifting and changing where is now is holding information that describes where it was from and what it means for this item to be completed or not completed.

Reply:
This has been refactored and no longer exists